### DISCUSSION TITLE
Crash while busy uploading

### DISCUSSION THREAD

**Discussion by dantearmok** [transmission/transmission#5252] [Support]
Mac mini (late 2012, 2.5-GHz 4-Core i5), 16GB RAM, plus an assortment of USB & FW HDs.
macOS 10.15 Catalina, ClamXAV, PeakHour, and PIA (all fully updated).
Running Transmission (f5a24862db) -- March 17th's midnight nightly.

Transmission was happy doing its thing, then it crashed...

Process:               Transmission [29901]
Path:                  /Applications/Transmission.app/Contents/MacOS/Transmission
Identifier:            Transmission
Version:               4.0.2-dev (14714.4.0.2)
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           Transmission [29901]
User ID:               502

Date/Time:             2023-03-18 23:17:09.958 -0400
OS Version:            Mac OS X 10.15.7 (19H2026)
Report Version:        12
Anonymous UUID:        9D83084F-22F2-20FB-B0D2-056E94905F2F

Time Awake Since Boot: 79000 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000001160
Exception Note:        EXC_CORPSE_NOTIFY

Termination Signal:    Segmentation fault: 11
Termination Reason:    Namespace SIGNAL, Code 0xb
Terminating Process:   exc handler [29901]

VM Regions Near 0x1160:
--> 
    __TEXT                 000000010533a000-0000000105572000 [ 2272K] r-x/r-x SM=COW  /Applications/Transmission.app/Contents/MacOS/Transmission

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   org.m0k.transmission          	0x0000000105448951 HandshakeMediator::is_peer_known_seed(int, tr_address const&) const + 11
1   org.m0k.transmission          	0x0000000105445666 tr_peerMgrPeerStats(tr_torrent const*, unsigned long*) + 308
2   org.m0k.transmission          	0x000000010537cb2e -[Torrent peers] + 59
3   org.m0k.transmission          	0x00000001053cfa84 -[InfoPeersViewController updateInfo] + 1087
4   org.m0k.transmission          	0x0000000105396200 -[InfoWindowController updateInfoStats] + 48
5   org.m0k.transmission          	0x000000010534c35f -[Controller updateUI] + 922
6   com.apple.Foundation          	0x00007fff38493d6a __NSFireTimer + 67
7   com.apple.CoreFoundation      	0x00007fff35d8e718 __CFRUNLOOP_IS_CALLING_OUT_TO_A_TIMER_CALLBACK_FUNCTION__ + 20
8   com.apple.CoreFoundation      	0x00007fff35d8e27e __CFRunLoopDoTimer + 872
9   com.apple.CoreFoundation      	0x00007fff35d8dd69 __CFRunLoopDoTimers + 322
10  com.apple.CoreFoundation      	0x00007fff35d729ed __CFRunLoopRun + 1885
11  com.apple.CoreFoundation      	0x00007fff35d71c33 CFRunLoopRunSpecific + 466
12  com.apple.HIToolbox           	0x00007fff3498daad RunCurrentEventLoopInMode + 292
13  com.apple.HIToolbox           	0x00007fff3498d7c5 ReceiveNextEventCommon + 584
14  com.apple.HIToolbox           	0x00007fff3498d569 _BlockUntilNextEventMatchingListInModeWithFilter + 64
15  com.apple.AppKit              	0x00007fff32fd63c9 _DPSNextEvent + 883
16  com.apple.AppKit              	0x00007fff32fd4c10 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1352
17  com.apple.AppKit              	0x00007fff32fc691e -[NSApplication run] + 658
18  com.apple.AppKit              	0x00007fff32f98706 NSApplicationMain + 777
19  libdyld.dylib                 	0x00007fff6faf4cc9 start + 1

Thread 1:
0   libsystem_malloc.dylib        	0x00007fff6fcb01ac szone_malloc_should_clear + 296
1   libsystem_malloc.dylib        	0x00007fff6fcaed7a malloc_zone_malloc + 104
2   libsystem_malloc.dylib        	0x00007fff6fcaecf5 malloc + 21
3   libc++abi.dylib               	0x00007fff6ce29dea operator new(unsigned long) + 26
4   org.m0k.transmission          	0x000000010544d398 std::__1::__split_buffer<(anonymous namespace)::connect_helpers::peer_candidate, std::__1::allocator<(anonymous namespace)::connect_helpers::peer_candidate>&>::__split_buffer(unsigned long, unsigned long, std::__1::allocator<(anonymous namespace)::connect_helpers::peer_candidate>&) + 68
5   org.m0k.transmission          	0x0000000105447614 tr_peerMgr::makeNewPeerConnections(unsigned long) + 292
6   org.m0k.transmission          	0x00000001054471c3 tr_peerMgr::reconnectPulse() + 2915
7   org.m0k.transmission          	0x000000010544876b tr_peerMgr::bandwidthPulse() + 217
8   org.m0k.transmission          	0x00000001054d6c4b event_process_active_single_queue + 1121
9   org.m0k.transmission          	0x00000001054d3aeb event_base_loop + 1134
10  org.m0k.transmission          	0x0000000105434245 tr_session_thread_impl::sessionThreadFunc(event_base*) + 87
11  org.m0k.transmission          	0x0000000105434637 void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (tr_session_thread_impl::*)(event_base*), tr_session_thread_impl*, event_base*> >(void*) + 63
12  libsystem_pthread.dylib       	0x00007fff6fcfd109 _pthread_start + 148
13  libsystem_pthread.dylib       	0x00007fff6fcf8b8b thread_start + 15

Thread 2:
0   libsystem_kernel.dylib        	0x00007fff6fc3e0ee __select + 10
1   libcurl.4.dylib               	0x00007fff6d796b5b Curl_poll + 623
2   libcurl.4.dylib               	0x00007fff6d791af1 Curl_multi_wait + 609
3   org.m0k.transmission          	0x00000001054714ce tr_web::Impl::curlThreadFunc() + 414
4   org.m0k.transmission          	0x0000000105471764 void* std::__1::__thread_proxy<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, void (tr_web::Impl::*)(), tr_web::Impl*> >(void*) + 59
5   libsystem_pthread.dylib       	0x00007fff6fcfd109 _pthread_start + 148
6   libsystem_pthread.dylib       	0x00007fff6fcf8b8b thread_start + 15

Thread 3:
0   libsystem_kernel.dylib        	0x00007fff6fc3a756 kevent + 10
1   org.m0k.transmission          	0x00000001053e3172 -[VDKQueue watcherThread:] + 203
2   com.apple.Foundation          	0x00007fff3842d222 __NSThread__start__ + 1064
3   libsystem_pthread.dylib       	0x00007fff6fcfd109 _pthread_start + 148
4   libsystem_pthread.dylib       	0x00007fff6fcf8b8b thread_start + 15

Thread 4:: com.apple.NSEventThread
0   libsystem_kernel.dylib        	0x00007fff6fc35dea mach_msg_trap + 10
1   libsystem_kernel.dylib        	0x00007fff6fc36160 mach_msg + 60
2   com.apple.CoreFoundation      	0x00007fff35d73d07 __CFRunLoopServiceMachPort + 247
3   com.apple.CoreFoundation      	0x00007fff35d727c1 __CFRunLoopRun + 1329
4   com.apple.CoreFoundation      	0x00007fff35d71c33 CFRunLoopRunSpecific + 466
5   com.apple.AppKit              	0x00007fff33177bf4 _NSEventThread + 132
6   libsystem_pthread.dylib       	0x00007fff6fcfd109 _pthread_start + 148
7   libsystem_pthread.dylib       	0x00007fff6fcf8b8b thread_start + 15

Thread 5:: com.apple.coreaudio.AQClient
0   libsystem_kernel.dylib        	0x00007fff6fc35dea mach_msg_trap + 10
1   libsystem_kernel.dylib        	0x00007fff6fc36160 mach_msg + 60
2   com.apple.CoreFoundation      	0x00007fff35d73d07 __CFRunLoopServiceMachPort + 247
3   com.apple.CoreFoundation      	0x00007fff35d727c1 __CFRunLoopRun + 1329
4   com.apple.CoreFoundation      	0x00007fff35d71c33 CFRunLoopRunSpecific + 466
5   com.apple.audio.toolbox.AudioToolbox	0x00007fff340e8f60 GenericRunLoopThread::Entry(void*) + 166
6   libAudioToolboxUtility.dylib  	0x00007fff6c47c611 CAPThread::Entry(CAPThread*) + 77
7   libsystem_pthread.dylib       	0x00007fff6fcfd109 _pthread_start + 148
8   libsystem_pthread.dylib       	0x00007fff6fcf8b8b thread_start + 15

Thread 6:
0   libsystem_kernel.dylib        	0x00007fff6fc35e26 semaphore_wait_trap + 10
1   com.apple.audio.caulk         	0x00007fff69638b16 caulk::mach::semaphore::wait() + 16
2   com.apple.audio.caulk         	0x00007fff696389b2 caulk::semaphore::timed_wait(double) + 106
3   com.apple.audio.caulk         	0x00007fff696387c4 caulk::concurrent::details::worker_thread::run() + 30
4   com.apple.audio.caulk         	0x00007fff696381e4 void* caulk::thread_proxy<std::__1::tuple<caulk::thread::attributes, void (caulk::concurrent::details::worker_thread::*)(), std::__1::tuple<caulk::concurrent::details::worker_thread*> > >(void*) + 45
5   libsystem_pthread.dylib       	0x00007fff6fcfd109 _pthread_start + 148
6   libsystem_pthread.dylib       	0x00007fff6fcf8b8b thread_start + 15

Thread 7:: AMCP Logging Spool
0   libsystem_kernel.dylib        	0x00007fff6fc35e26 semaphore_wait_trap + 10
1   com.apple.audio.caulk         	0x00007fff69638b16 caulk::mach::semaphore::wait() + 16
2   com.apple.audio.caulk         	0x00007fff696389b2 caulk::semaphore::timed_wait(double) + 106
3   com.apple.audio.caulk         	0x00007fff696387c4 caulk::concurrent::details::worker_thread::run() + 30
4   com.apple.audio.caulk         	0x00007fff696381e4 void* caulk::thread_proxy<std::__1::tuple<caulk::thread::attributes, void (caulk::concurrent::details::worker_thread::*)(), std::__1::tuple<caulk::concurrent::details::worker_thread*> > >(void*) + 45
5   libsystem_pthread.dylib       	0x00007fff6fcfd109 _pthread_start + 148
6   libsystem_pthread.dylib       	0x00007fff6fcf8b8b thread_start + 15

Thread 8:
0   libsystem_pthread.dylib       	0x00007fff6fcf8b68 start_wqthread + 0

Thread 9:
0   libsystem_pthread.dylib       	0x00007fff6fcf8b68 start_wqthread + 0

Thread 10:
0   libsystem_pthread.dylib       	0x00007fff6fcf8b68 start_wqthread + 0

Thread 11:
0   libsystem_pthread.dylib       	0x00007fff6fcf8b68 start_wqthread + 0

Thread 12:
0   libsystem_pthread.dylib       	0x00007fff6fcf8b68 start_wqthread + 0

Thread 0 crashed with X86 Thread State (64-bit):
  rax: 0x0000000019d25800  rbx: 0x00007fb41a0d5e00  rcx: 0x0000000000000010  rdx: 0xffffffffffffffd0
  rdi: 0x0000000000000000  rsi: 0x00007fb419d25800  rbp: 0x00007ffeea8c2ea0  rsp: 0x00007ffeea8c2ea0
   r8: 0x00000000000007d0   r9: 0x0000000000000000  r10: 0x0000000000000007  r11: 0xffffffb52fc1fc30
  r12: 0x00007fb419d25800  r13: 0x00007ffeea8c2eb8  r14: 0x0000600000392598  r15: 0x0000000064167e79
  rip: 0x0000000105448951  rfl: 0x0000000000010287  cr2: 0x0000000000001160
  
Logical CPU:     1
Error Code:      0x00000004 (no mapping for user data read)
Trap Number:     14

**Comment by ckerr** [comment]
Moved to bug reports @ https://github.com/transmission/transmission/issues/5256
